### PR TITLE
Convert HTML entities to strings (#4125)

### LINKED ISSUE #4125: JSX: Convert HTML entities into UTF-8
[HTML entities](https://facebook.github.io/react/docs/jsx-gotchas.html#html-entities) within JSX-style-nodes e.g. `<div>&nbsp;</div>` are currently not converted into their UTF-8 representation (e.g. [how babel does it](https://github.com/babel/babel/blob/b91b4a4dc03eff8647b669c8935b18f0c6d7ed81/packages/babylon/src/plugins/jsx/index.js#L62)). Are you planning to implement this?

### PR DESCRIPTION
**Pull request by RyanCavanaugh**
Fixes #4125. HTML entity list from Wikipedia https://en.wikipedia.org/wiki/List_of_XML_and_HTML_character_entity_references

### CODE REVIEW

File: src/compiler/emitter.ts [hunk 1]
```diff
@@ -7139,4 +7152,260 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
             }
         }
     }
+
+    var entities: Map<number> = {
```
Comment (mhegazy) [inline src/compiler/emitter.ts]: const

**Comment by mhegazy** [comment]
:+1:

File: src/compiler/emitter.ts (hunk 1 above)
Comment (RyanCavanaugh) [inline src/compiler/emitter.ts]: Can't use `const` here until we fix the bug about TDZ and functions

**Comment by huan086** [comment]
This doesn't handle the ones with &#

**Pull request merged**